File: web/src/services/authApi.ts
import api from './api';
import type { AuthResponse, LoginRequest, RegisterRequest, ResetPasswordRequest } from '../types';

export const authApi = {
  /**
   * Register a new user
   */
  async register(email: string, password: string, timezone?: string): Promise<AuthResponse> {
    const request: RegisterRequest = {
      email,
      password,
      timezone: timezone || Intl.DateTimeFormat().resolvedOptions().timeZone || 'UTC',
    };
    const response = await api.post<AuthResponse>('/auth/register', request);
    return response.data;
  },

  /**
   * Login with email and password
   */
  async login(email: string, password: string): Promise<AuthResponse> {
    const request: LoginRequest = { email, password };
    const response = await api.post<AuthResponse>('/auth/login', request);
    return response.data;
  },

  /**
   * Request a password reset token
   */
  async requestPasswordReset(email: string): Promise<{ message: string; token: string }> {
    const response = await api.post<{ message: string; token: string }>(
      '/auth/reset-password-request',
      { email }
    );
    return response.data;
  },

  /**
   * Reset password with token
   */
  async resetPassword(token: string, newPassword: string): Promise<{ message: string }> {
    const request: ResetPasswordRequest = { token, newPassword };
    const response = await api.post<{ message: string }>('/auth/reset-password', request);
    return response.data;
  },

  /**
   * Verify email address with token
   */
  async verifyEmail(token: string): Promise<{ message: string }> {
    const response = await api.get<{ message: string }>(`/auth/verify-email?token=${token}`);
    return response.data;
  },
};

File: mobile/lib/services/auth_api.dart
import '../models/auth_response.dart';
import 'api_service.dart';

class AuthApi {
  final ApiService _apiService;

  static AuthApi get instance => authApi;

  AuthApi(this._apiService);

  /// Register a new user
  Future<AuthResponse> register({
    required String email,
    required String password,
    String? timezone,
  }) async {
    final response = await _apiService.post(
      '/auth/register',
      data: {
        'email': email,
        'password': password,
        'timezone': timezone ?? 'UTC',
      },
    );

    return AuthResponse.fromJson(response.data as Map<String, dynamic>);
  }

  /// Login with email and password
  Future<AuthResponse> login({
    required String email,
    required String password,
  }) async {
    final response = await _apiService.post(
      '/auth/login',
      data: {
        'email': email,
        'password': password,
      },
    );

    return AuthResponse.fromJson(response.data as Map<String, dynamic>);
  }

  /// Request password reset
  Future<void> requestPasswordReset({required String email}) async {
    await _apiService.post(
      '/auth/reset-password-request',
      data: {
        'email': email,
      },
    );
  }

  /// Reset password with token
  Future<void> resetPassword({
    required String token,
    required String newPassword,
  }) async {
    await _apiService.post(
      '/auth/reset-password',
      data: {
        'token': token,
        'newPassword': newPassword,
      },
    );
  }

  /// Verify email address with token
  Future<Map<String, dynamic>> verifyEmail({required String token}) async {
    final response = await _apiService.get(
      '/auth/verify-email',
      queryParameters: {
        'token': token,
      },
    );

    return response.data as Map<String, dynamic>;
  }
}

// Singleton instance
final authApi = AuthApi(apiService);
